compiler/parser/RdrHsSyn.hs

```diff
@@ -2533,32 +2533,53 @@ data TemplateBodyDecl
   | KeyDecl (Located KeyData)
   | MaintainerDecl (LHsExpr GhcPs)
 
+data TemplateBodyDecls =
+  TemplateBodyDecls {
+      tbdEnsures :: [LHsExpr GhcPs]
+    , tbdSignatories :: [LHsExpr GhcPs]
+    , tbdObservers :: [LHsExpr GhcPs]
+    , tbdAgreements :: [LHsExpr GhcPs]
+    , tbdControlledChoiceGroups :: [Located (LHsExpr GhcPs, Located [Located ChoiceData])]
+    , tbdLetBindings :: [LHsLocalBinds GhcPs]
+    , tbdFlexChoices :: [Located FlexChoiceData]
+    , tbdKeys :: [Located KeyData]
+    , tbdMaintainers :: [LHsExpr GhcPs]
+    }
+
+-- I don't believe I can derive 'Monoid' without adding a package dependency.
+
+instance Semigroup TemplateBodyDecls where
+  (<>) x y = TemplateBodyDecls {
+      tbdEnsures = tbdEnsures x Monoid.<> tbdEnsures y
+    , tbdSignatories = tbdSignatories x Monoid.<> tbdSignatories y
+    , tbdObservers = tbdObservers x Monoid.<> tbdObservers y
+    , tbdAgreements = tbdAgreements x Monoid.<> tbdAgreements y
+    , tbdControlledChoiceGroups = tbdControlledChoiceGroups x Monoid.<> tbdControlledChoiceGroups y
+    , tbdLetBindings = tbdLetBindings x Monoid.<> tbdLetBindings y
+    , tbdFlexChoices = tbdFlexChoices x Monoid.<> tbdFlexChoices y
+    , tbdKeys = tbdKeys x Monoid.<> tbdKeys y
+    , tbdMaintainers = tbdMaintainers x Monoid.<> tbdMaintainers y
```

Comment:
Simply `<>` should be sufficient. It's in Prelude nowadays.

Reply:
Conflict with `Outputable` in this module. That's why the pre-existing uses are qualified also.

Reply:
Ah, Outputable, that really should be a Monoid or not define <> - but yeah, not something to tackle in our fork.